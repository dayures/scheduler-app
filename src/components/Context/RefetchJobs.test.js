import React from 'react'
import { shallow } from 'enzyme'
import RefetchJobsContext from './RefetchJobs'

describe('RefetchJobsContext', () => {
    it('exports an object with provider and consumer', () => {
        expect('Provider' in RefetchJobsContext).toBe(true)
        expect('Consumer' in RefetchJobsContext).toBe(true)
    })
})

describe('RefetchJobsContext.Consumer', () => {
    /**
     * This test would probably be more straightforward if we used react testing
     * library (https://testing-library.com). But since we're using enzyme here
     * I stuck with that, to avoid having two test approaches in this codebase.
     */

    it('returns a function that throws an error if used outside of the provider', () => {
        const createWrapper = () =>
            shallow(
                <RefetchJobsContext.Consumer>
                    {refetch => refetch()}
                </RefetchJobsContext.Consumer>
            )

        expect(createWrapper).toThrowErrorMatchingSnapshot()
    })
})
