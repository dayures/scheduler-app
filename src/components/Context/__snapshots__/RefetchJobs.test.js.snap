// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefetchJobsContext.Consumer returns a function that throws an error if used outside of the provider 1`] = `"RefetchJobsContext consumer needs to have a valid Provider as parent"`;
